tests/callbacks/test_RecorderListener.cpp

```diff
@@ -78,7 +78,7 @@ TYPED_TEST(RecorderListenerSingleEventTypeTest, record)
 //! @test Check the behavior of @ref Kokkos::utils::callbacks::RecorderListener.
 TEST_F(RecorderListenerTest, recorded_events)
 {
-    EventInProfileSectionRegexMatcher matcher(std::regex("profile section"));
+    EventInProfileSectionMatcher matcher{.matcher = EventRegexMatcher{.regex = std::regex("profile section")}};
```

Comment:
OK! That's interesting. Didn't know that would work without specifying the template parameter. Looks like it's a C++20 feature.